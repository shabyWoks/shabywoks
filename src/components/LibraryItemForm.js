import React from 'react';

class LibraryItemForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            id: this.props.libraryItem ? this.props.libraryItem.id : -1,
            tag: this.props.libraryItem ? this.props.libraryItem.tag : '',
            name: this.props.libraryItem ? this.props.libraryItem.name : '',
            description: this.props.libraryItem ? this.props.libraryItem.description : '',
            docsLink: this.props.libraryItem? this.props.libraryItem.docsLink : ''
        }
    }
    onTagChange = (e) => {
        const tag = e.target.value;
        this.setState(() => ({ tag }));
    }
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    }
    onNameChange = (e) => {
        const name = e.target.value;
        this.setState(() => ({ name }));
    }
    onDocsLinkChange = (e) => {
        const docsLink = e.target.value;
        this.setState(() => ({ docsLink }));
    }
    onClick = () => {
        this.props.onSubmit(this.state);
    }
    render() {
        return (
            <div>
                Tag Name: <input type="text" value={this.state.tag} onChange={this.onTagChange}/> <br />
                Name: <input type="text" value={this.state.name} onChange={this.onNameChange}/> <br />
                Description: <input type="text" value={this.state.description} onChange={this.onDescriptionChange}/> <br />
                Docs Link: <input type="text" value={this.state.docsLink} onChange={this.onDocsLinkChange}/> <br />
                <button onClick= {this.onClick}>ADD</button>
            </div>
        );
    }
}

export default LibraryItemForm;