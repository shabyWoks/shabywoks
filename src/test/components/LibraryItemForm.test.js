import React from 'react';
import {shallow} from 'enzyme';
import LibraryItemForm from '../../components/LibraryItemForm';
import libraryItems from '../fixtures/libraryItems';

let onSubmit, wrapper;

beforeEach(() => {
    onSubmit = jest.fn();
    wrapper = shallow(<LibraryItemForm libraryItem={libraryItems[0]} onSubmit={onSubmit}/>);
})

test('should have rendered library item form correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle tag change correctly', () => {
    const value= "New Value";
    wrapper.find('input').at(0).simulate('change', {
        target: {value: value }
    });
    expect(wrapper.state('tag')).toBe(value);
});

test('should handle name change correctly', () => {
    const value= "New Value";
    wrapper.find('input').at(1).simulate('change', {
        target: {value: value }
    });
    expect(wrapper.state('name')).toBe(value);
});

test('should handle description change correctly', () => {
    const value= "New Value";
    wrapper.find('input').at(2).simulate('change', {
        target: {value: value }
    });
    expect(wrapper.state('description')).toBe(value);
});

test('should handle docsLink change correctly', () => {
    const value= "New Value";
    wrapper.find('input').at(3).simulate('change', {
        target: {value: value }
    });
    expect(wrapper.state('docsLink')).toBe(value);
});

test('should handle button click correctly', () => {
    wrapper.find('button').simulate('click');
    expect(onSubmit).toHaveBeenLastCalledWith(libraryItems[0]);
});