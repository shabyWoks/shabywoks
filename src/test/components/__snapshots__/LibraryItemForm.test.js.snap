// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have rendered library item form correctly 1`] = `
<div>
  Tag Name: 
  <input
    onChange={[Function]}
    type="text"
    value="React"
  />
   
  <br />
  Name: 
  <input
    onChange={[Function]}
    type="text"
    value="react-redux"
  />
   
  <br />
  Description: 
  <input
    onChange={[Function]}
    type="text"
    value="no description"
  />
   
  <br />
  Docs Link: 
  <input
    onChange={[Function]}
    type="text"
    value="no link"
  />
   
  <br />
  <button
    onClick={[Function]}
  >
    ADD
  </button>
</div>
`;
